### PR TITLE
Enable enforcing hotfix

### PR DESCRIPTION
**Pull request by julianpistorius**
## Description

Problem: Wrong parameter for enforcement function
    
`allocation_source_overage_enforcement_for_user` and all the subsequent
logic expects an allocation source object, but we pass in a string.
    
Solution: Pass in the `allocation_source` instead of
`allocation_source.name`

## Checklist before merging Pull Requests
- ~[ ] New test(s) included to reproduce the bug/verify the feature~
- ~[ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature~
- [ ] Reviewed and approved by at least one other contributor.
- ~[ ] If necessary, include a snippet in CHANGELOG.md~

### CODE REVIEW

**Comment by julianpistorius** [comment]
To be clear, enforcing will not happen on Jetstream without commit c66a09a. The naming for these tasks are not super-clear:

`monitor_jetstream_allocation_sources` vs `monitor_allocation_sources`

The first updates allocation sources based on the TAS API. The second one enforces allocation limits.

File: service/tasks/monitoring.py [hunk 1]
```diff
@@ -574,23 +574,31 @@ def monitor_allocation_sources(usernames=()):
...
             if usernames and user.username not in usernames:
                 celery_logger.info("Skipping User %s - not in the list" % user.username)
                 continue
             over_allocation = allocation_source.is_over_allocation(user)
+            celery_logger.debug('monitor_allocation_sources - user: %s, over_allocation: %s', user, over_allocation)
             if not over_allocation:
                 continue
-            allocation_source_overage_enforcement_for_user.apply_async(args=(allocation_source.name, user))
+            celery_logger.debug('monitor_allocation_sources - Going to enforce on user user: %s', user)
+            allocation_source_overage_enforcement_for_user.apply_async(args=(allocation_source, user))
```
Comment (steve-gregory) [inline service/tasks/monitoring.py]: Clarification: This is the actual fix to re-enable enforcing.

File: service/tasks/monitoring.py [hunk 2]
```diff
@@ -574,23 +574,31 @@ def monitor_allocation_sources(usernames=()):
     """
     Monitor allocation sources, if a snapshot shows that all compute has been used, then enforce as necessary
     """
+    celery_logger.debug('monitor_allocation_sources - usernames: %s', usernames)
     allocation_sources = AllocationSource.objects.all()
     for allocation_source in allocation_sources.order_by('name'):
-        for user in allocation_source.all_users:
+        celery_logger.debug('monitor_allocation_sources - allocation_source: %s', allocation_source)
+        for user in allocation_source.all_users.order_by('username'):
+            celery_logger.debug('monitor_allocation_sources - user: %s', user)
```
Comment (steve-gregory) [inline service/tasks/monitoring.py]: 👍  for Additional debug logging + Organized Allocations/Usernames!

File: atmosphere/settings/local.py.j2 [hunk 3]
```diff
@@ -155,39 +155,44 @@ CELERYBEAT_SCHEDULE["{{task_key}}"]["{{task_opt}}"] = {{task_opt_value}}
...
-        "options" : {"expires": 45 * 60,"time_limit": 45 * 60}
-  }
+        "options": {"expires": 45 * 60, "time_limit": 45 * 60}
+    },
+    "monitor_allocation_sources": {
+        "task": "monitor_allocation_sources",
+        # Every 15 minutes
+        "schedule": timedelta(minutes=15),
+        "options": {"expires": 15 * 60, "time_limit": 15 * 60}
+    }
```
Comment (steve-gregory) [inline atmosphere/settings/local.py.j2]: Clarification: The second portion of this hotfix is including `monitor_allocation_sources` in the `JETSTREAM_CELERYBEAT_SCHEDULE`

File: service/tasks/monitoring.py (hunk 1 above)
Comment (julianpistorius) [inline service/tasks/monitoring.py]: Yes - thanks for the clarification. 👍

File: atmosphere/settings/local.py.j2 (hunk 3 above)
Comment (julianpistorius) [inline atmosphere/settings/local.py.j2]: Thanks for pointing this out.

**Pull request merged**